.github/workflows/label-cherry-pick.yaml

```diff
@@ -75,41 +76,59 @@ jobs:
+
           set +e
-          git cherry-pick -x $MERGE_COMMIT_SHA
+          # Select the first parent (-m 1) as the mainline tree. This is necessary if
+          # the commit has multiple parents as the cherry-pick will fail otherwise.
+          git cherry-pick -x -m 1 ${MERGE_COMMIT_SHA}
           RES=$?
           set -e
-          if [ $RES -ne 0 ]; then
+          if [ ${RES} -ne 0 ]; then
+            # If the cherry pick failed due to a merge conflict we can
+            # add the conflicting file and create the commit anyway.
             git add .
             git cherry-pick --continue
           fi
-          exit $RES
+          exit ${RES}
 
       - name: Create Pull Request
         id: create-pr
+        continue-on-error: true
         uses: peter-evans/create-pull-request@2b011faafdcbc9ceb11414d64d0573f37c774b04 # v4.2.3
         with:
           token: ${{ secrets.CHERRY_PICK_TOKEN }}
           draft: ${{ steps.cherry-pick.outcome == 'failure' }}
           base: ${{ matrix.target_branch }}
-          branch: "${{ matrix.target_branch }}-${{ github.event.pull_request.number }}"
+          branch: "cherry-pick-${{ matrix.target_branch }}-${{ github.event.pull_request.number }}"
           committer: GitHub Release Automation <github@google.com>
           reviewers: ${{ github.event.pull_request.user.login }}
           title: "Cherry pick PR #${{ github.event.pull_request.number }}: ${{ github.event.pull_request.title }}"
           body: |
             Refer to the original PR: https://github.com/${{ github.repository }}/pull/${{ github.event.pull_request.number }}
 
             ${{ github.event.pull_request.body }}
-      - name: PR Comment
-        if: ${{ success() && steps.cherry-pick.outcome == 'failure' }}
-        uses: actions/github-script@v2
+
+      - name: Comment on failure
+        uses: actions/github-script@v6
         with:
           github-token: ${{ secrets.CHERRY_PICK_TOKEN }}
           script: |
-            github.issues.createComment({
-              issue_number: ${{ steps.create-pr.outputs.pull-request-number }},
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              body: '> [!IMPORTANT]\n> The cherry-pick failed! Check out the PR branch (${{ matrix.target_branch }}-${{ github.event.pull_request.number }}) and fix the conflicts before proceeding.'
-            })
+            if ('${{ steps.create-pr.outputs.pull-request-number }}' == '') {
+              // Comment on the originating PR if creating a cherry pick PR failed.
+              github.rest.issues.createComment({
+                issue_number: context.payload.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                body: '> [!IMPORTANT]\n> Creating the cherry pick PR failed! Check the log at ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }} for details.'
```

Comment:
Is it necessary to but github vars like github.repository in env vars like in other workflows?

Reply:
No. It's good practice to do so if the variable is or can be user generated as that can allow remote code execution.
The variables I used here are not user generated and the workflow runs on `main` so the risk for this is non-existent.

Reply:
Thanks, that distinction makes sense